### DISCUSSION TITLE
WARN main [AdjointIntegrator] render()

### DISCUSSION THREAD

**Discussion by 774880930** [mitsuba-renderer/mitsuba3#382] [Q&A] [answered]
Hello,when I  learn the caustic optimization and  render a picture,there is a warnning,but i can't find where to add it,warnning is here:
WARN main [AdjointIntegrator] render(): forcing 'evaluate=true' since multi-pass rendering was requested.
Where could I to force it?

**Comment by njroussel** [comment, marked as answer]
There is nothing to do on your side. The warning is telling you that it had to change/force `evaluate` to tTrue` otherwise it would not have been able to continue. This usually happens if you're rendering with a high sample count per pixel.
If you're using `mi.render`, unfortunately, there is no way to set `evaluate` on your end,  it is set to `False` internally.

**Reply by 774880930** [reply]
I got it.Thanks for your reply.
